pydocstringformatter/_formatting/formatters_numpydoc.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+from pydocstringformatter._formatting.base import NumpydocSectionFormatter
+
+
+class NumpydocSectionOrderingFormatter(NumpydocSectionFormatter):
+    """Change section order to match numpydoc guidelines."""
+
+    name = "numpydoc-section-order"
+
+    numpydoc_section_order = (
+        # I'm folding summary, deprecation warning, and extended
+        # summary into "before first heading"
+        "Parameters",
+        "Attributes",
+        "Methods",
+        "Returns",
+        "Yields",
+        "Receives",
+        "Other Parameters",
+        "Raises",
+        "Warns",
+        "Warnings",
+        "See Also",
+        "Notes",
+        "References",
+        "Examples",
+    )
+
+    def treat_sections(self, sections: list[list[str]]) -> list[list[str]]:
+        """Sort the numpydoc sections into the numpydoc order."""
+        # section_names = [sec_lines[0].strip() for sec_lines in sections]
+        # assert all(name in self.numpydoc_section_order for name in section_names[1:])
+        new_sections = sorted(
+            sections[1:],
+            key=lambda sec_lines: self.numpydoc_section_order.index(
+                sec_lines[0].strip()
+            ),
+        )
+        new_sections.insert(0, sections[0])
+        return new_sections
+
+
+class NumpydocParameterNameColonTypeFormatter(NumpydocSectionFormatter):
```

Comment:
```suggestion
class NumpydocColonTypeFormatter(NumpydocSectionFormatter):
```

Reply:
What usually throws me off is the space between the parameter name and colon, not between the colon and the type (I write `x: float` instead of `x : float`; `sphinxcontrib.ext.napolean` accepts both; `numpydoc` accepts only the second).  Maybe 
```suggestion
class NumpydocNameColonTypeFormatter(NumpydocSectionFormatter):
```
since it also applies to the returned values?

Reply:
Yeah that works for me!